python/ray/external_storage.py

```diff
@@ -1,8 +1,98 @@
 import abc
 import os
+import re
+from collections import namedtuple
 from typing import List
 
 import ray
+from ray.ray_constants import DEFAULT_OBJECT_PREFIX
+
+
+class ExternalStorageURLProtocol:
+    """Class to parse/generate urls for object spilling.
+
+    There will be 2 types of URLs.
```

Comment:
I think the concept of two types of URLs is confusing to understand. How about we call them plain URLWithOffset and BaseURL?

Then you have 
`get_base_url(url_with_offset) -> base_url`

Reply:
Are they conceptually different from internal/external URLs? Or is it just the naming change?

Reply:
Just naming convention